res/controllers/Denon-MC7000-scripts.js

```diff
@@ -561,39 +701,143 @@ MC7000.PadButtons = function(channel, control, value, status, group) {
+                if (engine.getValue("[Sampler" + samplerOffset + "]", "track_loaded") === 0) {
+                    engine.setValue("[Sampler" + samplerOffset + "]", "LoadSelectedTrack", 1);
+                } else if (engine.getValue("[Sampler" + samplerOffset + "]", "track_loaded") === 1) {
                     if (MC7000.prevSamplerStop) {
-                        // stop playing all other samplers ...
-                        for (j = 1; j <=8; j++) {
-                            engine.setValue("[Sampler" + j + "]", "cue_gotoandstop", 1);
+                        // stop playing all other samplers on this deck
+                        for (j = 1; j <= 8; j++) {
+                            samplerOffsetJ = deckOffset * 8 + j;
+                            if (engine.getValue("[Sampler" + samplerOffsetJ + "]", "play") === 1) {  // if sampler is playing then stop it
+                                engine.setValue("[Sampler" + samplerOffsetJ + "]", "cue_gotoandstop", 1);
+                            }
                         }
                     }
                     // ... before the actual sampler to play gets started
-                    engine.setValue("[Sampler" + i + "]", "cue_gotoandplay", 1);
+                    engine.setValue("[Sampler" + samplerOffset + "]", "pregain", 1);
+                    engine.setValue("[Sampler" + samplerOffset + "]", "cue_gotoandplay", 1);
                 }
-            } else if (control === 0x1C + i - 1 && value >= 0x01) {
-                if (engine.getValue("[Sampler" + i + "]", "play") === 1) {
-                    engine.setValue("[Sampler" + i + "]", "cue_gotoandstop", 1);
+            } else if (control === 0x1C + i - 1 && value >= 0x01) { //shifted button deactivates playing sampler or ejects sampler
+                if (MC7000.SamplerQty === 16) {
+                    if ((deckNumber - 1) >= 2) {
+                        deckOffset = deckOffset + 2;
+                    }
+                }
+                if (engine.getValue("[Sampler" + samplerOffset + "]", "play") === 1) {
+                    engine.setValue("[Sampler" + samplerOffset + "]", "cue_gotoandstop", 1);
                     midi.sendShortMsg(0x94 + deckOffset, 0x1C + i - 1, MC7000.padColor.samplerloaded);
                 } else {
-                    engine.setValue("[Sampler" + i + "]", "eject", 1);
+                    engine.setValue("[Sampler" + samplerOffset + "]", "eject", 1);
                     midi.sendShortMsg(0x94 + deckOffset, 0x1C + i - 1, MC7000.padColor.sampleroff);
-                    engine.setValue("[Sampler" + i + "]", "eject", 0);
+                    engine.setValue("[Sampler" + samplerOffset + "]", "eject", 0);
                 }
             }
         }
-    } else if (MC7000.PADModeVelSamp[deckNumber]) {
-        return;
-    } else if (MC7000.PADModePitch[deckNumber]) {
-        return;
-    }
-};
+    } else if (MC7000.PADModeVelSamp[deckNumber]) {  // reset pregain on stop required?
```

Comment:
I added some lines regarding this topic anyway, to avoid misbehavior when playing velocity samplers with the controller and samplers on mixxx user interface at the same time.